### PR TITLE
Add Capitual payment method

### PR DESCRIPTION
**Pull request by BtcContributor**
Implement https://github.com/bisq-network/growth/issues/246

Waiting for final Info text to be suggested by @pazza83

In the meantime, if you have any feedback on this please let me know :)

capitual

### CODE REVIEW

**Comment by BtcContributor** [comment]
@pazza83 do you have any update on final info text to add? Thanks!

**Comment by pazza83** [comment]
Hi @BtcContributor still waiting for my feedback on this [comment](https://github.com/bisq-network/growth/issues/246#issuecomment-851571559) from @m52go 

I do not think any special information is needed.  Let me know if you think I am overlooking anything

File: core/src/main/java/bisq/core/locale/CurrencyUtil.java [hunk 1]
```diff
@@ -347,6 +347,17 @@ public static void setBaseCurrencyCode(String baseCurrencyCode) {
...
 
+    public static List<TradeCurrency> getAllCapitualCurrencies() {
+        List<TradeCurrency> currencies = new ArrayList<>(Arrays.asList(
+                new FiatCurrency("EUR"),
+                new FiatCurrency("BRL"),
+                new FiatCurrency("GBP"),
+                new FiatCurrency("USD")
+        ));
+        currencies.sort(Comparator.comparing(TradeCurrency::getCode));
+        return currencies;
```
Comment (kaiwolfram) [inline core/src/main/java/bisq/core/locale/CurrencyUtil.java]: Sorting is not necessary when you initialize it in the correct order
```suggestion
        return new ArrayList<>(Arrays.asList(
                new FiatCurrency("BRL"),
                new FiatCurrency("EUR"),
                new FiatCurrency("GBP"),
                new FiatCurrency("USD")
        ));
```

File: desktop/src/main/java/bisq/desktop/components/paymentmethods/CapitualForm.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        FlowPane flowPane = labelFlowPaneTuple2.second;
+
+        if (isEditable)
+            flowPane.setId("flow-pane-checkboxes-bg");
+        else
+            flowPane.setId("flow-pane-checkboxes-non-editable-bg");
+
+        CurrencyUtil.getAllCapitualCurrencies().stream().forEach(e ->
+                fillUpFlowPaneWithCurrencies(isEditable, flowPane, e, capitualAccount));
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/components/paymentmethods/CapitualForm.java]: You can use `forEach` on the list itself
```suggestion
        CurrencyUtil.getAllCapitualCurrencies().forEach(e ->
                fillUpFlowPaneWithCurrencies(isEditable, flowPane, e, capitualAccount));
```

File: desktop/src/main/java/bisq/desktop/components/paymentmethods/CapitualForm.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                capitualAccount.getAccountNr());
+        addLimitations(true);
+        addCurrenciesGrid(false);
+    }
+
+    @Override
+    public void updateAllInputsValid() {
+        allInputsValid.set(isAccountNameValid()
+                && capitualValidator.validate(capitualAccount.getAccountNr()).isValid
+                && capitualAccount.getTradeCurrencies().size() > 0);
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/components/paymentmethods/CapitualForm.java]:
```suggestion
                && !capitualAccount.getTradeCurrencies().isEmpty());
```
It's faster and makes your intent clearer

Comment (ripcurlx) [reply desktop/src/main/java/bisq/desktop/components/paymentmethods/CapitualForm.java]: @KaiWitt Thanks for starting to do code reviews :+1:

**Comment by BtcContributor** [comment]
Applied feedbacks from @KaiWitt (thanks!) and added text suggestion from @pazza83 

I think now it is ready to be moved to final PR and no draft anymore :)

File: core/src/main/resources/i18n/displayStrings.properties [hunk 4]
```diff
@@ -3466,7 +3467,7 @@ payment.amazonGiftCard.info=To pay with Amazon eGift Card, you will need to send
   - try to use creative, believable text for the gift card''s message (e.g., "Happy birthday Susan!") along with the trade ID (and use trader chat \
   to tell your trading peer the reference text you picked so they can verify your payment)\n\
   - Amazon eGift Cards can only be redeemed on the Amazon website they were purchased on (e.g., a gift card purchased on amazon.it can only be redeemed on amazon.it)
-
+payment.capitual.info=Include any special instructions for the payment method that both buyer and seller might need to be aware of
```
Comment (m52go) [inline core/src/main/resources/i18n/displayStrings.properties]: FYI this line actually shows in the software...it should probably be removed.

Screenshot from 2021-06-15 23-48-32

Comment (BtcContributor) [reply core/src/main/resources/i18n/displayStrings.properties]: So, no popup or different text?

Comment (m52go) [reply core/src/main/resources/i18n/displayStrings.properties]: I would say if there's no text worth putting there, there should be no pop-up.

Comment (pazza83) [reply core/src/main/resources/i18n/displayStrings.properties]: Yes no special text needed.

**Comment by BtcContributor** [comment]
@ripcurlx @pazza83 now it should be good, popup removed

**Comment by ripcurlx** [comment]
@BtcContributor I completely missed this PR for the v1.7.2 release. Is this PR ready to be reviewed and tested?

**Comment by ripcurlx** [comment]
> @BtcContributor I completely missed this PR for the v1.7.2 release. Is this PR ready to be reviewed and tested?

Ok - I'll test and cherry-pick it for the release if everything looks fine tomorrow.

File: core/src/main/resources/i18n/displayStrings.properties (hunk 4 above)
Comment (ripcurlx) [reply core/src/main/resources/i18n/displayStrings.properties]: Please remove this unused translation.

**Review by ripcurlx — APPROVED** [review]
ACK (with my last two commits)

Bildschirmfoto 2021-07-19 um 13 52 33
Bildschirmfoto 2021-07-19 um 13 52 18

I assume the CAP code is sufficient for trading @pazza83 ?

**Pull request merged**